File: ddpoliglot_v6_flutter/lib/models/data/ArticleActor.dart
// ignore_for_file: file_names
import './Article.dart';

import 'package:json_annotation/json_annotation.dart';

part 'ArticleActor.g.dart';

@JsonSerializable()
class ArticleActor {
  int articleActorID;
  String? keyString;
  String? name;
  int role; // 0 text speach, 1- dictor voice
  bool defaultInRole;
  String? language;
  String? voiceName;
  double voiceSpeed;
  double voicePitch;
  int articleID;
  Article? article;

  ArticleActor(
    this.articleActorID,
    this.keyString,
    this.name,
    this.role, // 0 text speach, 1- dictor voice
    this.defaultInRole,
    this.language,
    this.voiceName,
    this.voiceSpeed,
    this.voicePitch,
    this.articleID,
    this.article,
  );

  factory ArticleActor.fromJson(Map<String, dynamic> data) =>
      _$ArticleActorFromJson(data);

  Map<String, dynamic> toJson() => _$ArticleActorToJson(this);
}

File: ddPoliglotV6.Data/Models/ArticleActor.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ddPoliglotV6.Data.Models
{
    public class ArticleActor
    {
        public int ArticleActorID { get; set; }
        public Guid KeyGuid { get; set; }
        [MaxLength(200)]
        public string Name { get; set; }
        public int Role { get; set; } // 0 text speach, 1- dictor voice
        public bool DefaultInRole { get; set; }
        [MaxLength(10)]
        public string Language { get; set; }
        [MaxLength(50)]
        public string VoiceName { get; set; }
        public decimal VoiceSpeed { get; set; }
        public decimal VoicePitch { get; set; }

        public int ArticleID { get; set; }
        [ForeignKey("ArticleID")]
        public Article Article { get; set; }
    }
}
